include/Addon/Upseller.php

```diff
@@ -0,0 +1,95 @@
+<?php namespace EmailLog\Addon;
+
+use EmailLog\Core\Loadie;
+
+defined( 'ABSPATH' ) || exit; // Exit if accessed directly.
+
+/**
+ * Upsells add-ons by displaying links to add-ons with context in different parts of admin interface.
+ *
+ * @since 2.4.0
+ */
+class Upseller implements Loadie {
+
+	// phpcs:ignore Squiz.Commenting.FunctionComment.Missing
+	public function load() {
+		add_action( 'admin_init', [ 'PAnD', 'init' ] );
+
+		add_action( 'el_before_logs_list_table', [ $this, 'upsell_auto_delete_logs_in_log_list_page' ] );
+		add_action( 'el_after_db_size_notification_setting', [ $this, 'upsell_auto_delete_logs_in_settings_page' ] );
+	}
+
+	/**
+	 * Renders Upsell message for Auto delete logs add-on in Log list page
+	 * if the number of logs is greater than 5000.
+	 *
+	 * @param int $total_logs Total number of logs.
+	 *
+	 */
+	public function upsell_auto_delete_logs_in_log_list_page( $total_logs ) {
+		if ( $total_logs < 5000 ) {
+			return;
+		}
+
+		if ( $this->is_addon_active( 'Auto Delete Logs' ) ) {
+			return;
+		}
+
+		if ( ! class_exists( 'PAnD' ) || ! \PAnD::is_admin_notice_active( 'disable-DL-upsell-notice-5000' ) ) {
+			return;
+		}
+		?>
+
+		<div data-dismissible="disable-DL-upsell-notice-5000" class="notice notice-warning is-dismissible">
+			<p>
+				<?php
+				/* translators: 1 Auto Delete Logs add-on name.  */
+				printf(
+					__( 'You have more than 5000 email logs in the database. You can use our %1s add-on to automatically delete logs as the DB size grows.', 'email-log' ),
+					'<a href="https://wpemaillog.com/addons/auto-delete-logs/?utm_campaign=Upsell&utm_medium=wpadmin&utm_source=log-list&utm_content=dl">Auto Delete Logs</a>'
+				);
+				?>
+			</p>
+		</div>
+
+		<?php
+	}
+
+	/**
+	 * Renders Upsell message for Auto delete logs add-on in Settings page.
+	 */
+	public function upsell_auto_delete_logs_in_settings_page() {
+		if ( $this->is_addon_active( 'Auto Delete Logs' ) ) {
+			return;
+		}
+
+		?>
+		<p>
+			<em>
+				<?php
+				printf(
+					__( 'You can also automatically delete logs if the database size increases using our <a href="https://wpemaillog.com/addons/auto-delete-logs/?utm_campaign=Upsell&utm_medium=wpadmin&utm_source=settings&utm_content=dl" target="_blank">Auto Delete Logs</a> add-on.', 'email-log' )
```

Comment:
You should not use html markup directly in `__` or `_e` functions. Check the above function to find out how to properly use `printf` function using format specifiers.